src/main/java/tutorlink/lists/GradeList.java

```diff
@@ -0,0 +1,64 @@
+package tutorlink.lists;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import tutorlink.exceptions.DuplicateGradeException;
+import tutorlink.exceptions.GradeNotFoundException;
+import tutorlink.exceptions.TutorLinkException;
+import tutorlink.grade.Grade;
+
+/**
+ * Represents a list of grades.
+ */
+public class GradeList {
+    private static final String ERROR_DUPLICATE_GRADE_ON_ADD = "Error! Grade (%s, %s) already exists in the list!";
+    private static final String ERROR_NO_GRADE_FOUND = "Error! Grade (%s, %s) does not exist in the list!";
+    private ArrayList<Grade> gradeArrayList;
+
+    public GradeList() {
+        this.gradeArrayList = new ArrayList<>();
+    }
+
+    public boolean deleteGrade(String matricNumber, String componentDescription) {
+        matricNumber = matricNumber.toUpperCase();
+        componentDescription = componentDescription.toLowerCase();
+        for (Grade grade : gradeArrayList) {
+            if (grade.getStudent().getMatricNumber().equals(matricNumber)
+                    && grade.getComponent().getName().equals(componentDescription)) {
+                return gradeArrayList.remove(grade);
+            }
+        }
+        return false;
+    }
+
+    public void addGrade(Grade grade) throws DuplicateGradeException {
+        for (Grade gradeToCompare : gradeArrayList) {
+            if (grade == gradeToCompare) {
```

Comment:
Use .equals or grade.isSameSubmission (which one to use depend command implementation)

 == in java only compared by references when using on non primitive types